pkg/k8s/report/report.go

```diff
@@ -168,34 +159,89 @@ func Write(report Report, option Option, securityChecks []string, showEmpty bool
 	}
 }
 
-func separateMisConfigRoleAssessment(k8sReport Report, securityChecks []string) (Report, Report) {
+type reports struct {
+	report  Report
+	columns []string
+}
+
+func separateMisconfigReports(k8sReport Report, securityChecks []string) []reports {
+
 	workloadMisconfig := make([]Resource, 0)
+	infraMisconfig := make([]Resource, 0)
 	rbacAssessment := make([]Resource, 0)
+
 	for _, misConfig := range k8sReport.Misconfigurations {
 		if slices.Contains(securityChecks, types.SecurityCheckRbac) && rbacResource(misConfig) {
 			rbacAssessment = append(rbacAssessment, misConfig)
-		} else {
-			if slices.Contains(securityChecks, types.SecurityCheckConfig) && !rbacResource(misConfig) {
-				workloadMisconfig = append(workloadMisconfig, misConfig)
+			continue
+		}
+
+		if infraResource(misConfig) {
+			workload, infra := splitInfraAndWorkloadResources(misConfig)
+
+			if slices.Contains(securityChecks, types.SecurityCheckK8SInfra) {
+				infraMisconfig = append(infraMisconfig, infra)
+			}
+
+			if slices.Contains(securityChecks, types.SecurityCheckConfig) {
+				workloadMisconfig = append(workloadMisconfig, workload)
 			}
+
+			continue
 		}
-	}
-	return Report{
-			SchemaVersion:     0,
-			ClusterName:       k8sReport.ClusterName,
-			Vulnerabilities:   k8sReport.Vulnerabilities,
-			Misconfigurations: workloadMisconfig,
-		}, Report{
-			SchemaVersion:     0,
-			ClusterName:       k8sReport.ClusterName,
-			Misconfigurations: rbacAssessment,
+
+		if slices.Contains(securityChecks, types.SecurityCheckConfig) && !rbacResource(misConfig) {
+			workloadMisconfig = append(workloadMisconfig, misConfig)
 		}
+	}
+
+	r := make([]reports, 0)
+
+	if slices.Contains(securityChecks, types.SecurityCheckConfig) && len(workloadMisconfig) > 0 {
+		r = append(r, reports{
+			report: Report{
+				SchemaVersion:     0,
+				ClusterName:       k8sReport.ClusterName,
+				Vulnerabilities:   k8sReport.Vulnerabilities,
+				Misconfigurations: workloadMisconfig,
+			},
+			columns: WorkloadColumns(),
+		})
+	}
+
+	if slices.Contains(securityChecks, types.SecurityCheckRbac) && len(rbacAssessment) > 0 {
+		r = append(r, reports{
+			report: Report{
+				SchemaVersion:     0,
+				ClusterName:       k8sReport.ClusterName,
+				Misconfigurations: rbacAssessment,
+			},
+			columns: RoleColumns(),
+		})
+	}
+
+	if slices.Contains(securityChecks, types.SecurityCheckK8SInfra) && len(infraMisconfig) > 0 {
+		r = append(r, reports{
+			report: Report{
+				SchemaVersion:     0,
+				ClusterName:       k8sReport.ClusterName,
+				Misconfigurations: infraMisconfig,
+			},
+			columns: InfraColumns(),
+		})
+	}
+
+	return r
 }
 
 func rbacResource(misConfig Resource) bool {
 	return misConfig.Kind == "Role" || misConfig.Kind == "RoleBinding" || misConfig.Kind == "ClusterRole" || misConfig.Kind == "ClusterRoleBinding"
 }
 
+func infraResource(misConfig Resource) bool {
+	return misConfig.Kind == "Pod" && misConfig.Namespace == "kube-system"
```

Comment:
I'm using rancher desktop and did not got infra result at all , hope our checks take  under consideration these use cases

Reply:
what do you see when you do `kubectl get pods -n kube-system` on rancher? (rancher is not working on my computer not sure why, but kind/minikube/kops/k3 are all based on kubeadm, which will have all k8s components under `kube-system`)

Reply:
```
NAME                                      READY   STATUS    RESTARTS   AGE
coredns-7448499f4d-4bsnw                  1/1     Running   3          11d
metrics-server-86cbb8457f-xtfv7           1/1     Running   3          11d
local-path-provisioner-5ff76fc89d-wzw7x   1/1     Running   5          11d
```

Reply:
So rancher doesn’t expose the apiserver? Controller manager? Or it is in another ns?